Feature/tests-environment-setup

javierriveros opened the pull request:
Initial configuration for the test environment.
The test environment is working now.
Github Actions is working.

javierriveros committed:
Add Docker files for setting up the test environment

javierriveros committed:
Add coverage config

javierriveros committed:
Add Makefile

javierriveros committed:
Add test app and a basic test

javierriveros committed:
Fix indentation in docker-compose.yml

javierriveros committed:
Remove settings env var from Dockerfile

javierriveros committed:
Change db connection port

javierriveros committed:
Move tests to testapp

javierriveros committed:
Move requirements to an independent folder

javierriveros committed:
Add tox.ini

javierriveros committed:
Setup Gihub actions

javierriveros committed:
Change manage.py path in Github actions

javierriveros committed:
Change database settings in actions

javierriveros committed:
Change database name in actions

javierriveros committed:
change database port in actions

javierriveros committed:
disable postgres service in actions

javierriveros committed:
change database settings for github actions env

javierriveros committed:
move testapp and manage.py to root folder

javierriveros committed:
include testapp in Dockerfile and change manage.py location in dev script

javierriveros committed:
set settings env variable for pytest

javierriveros committed:
change testapp location in github action

javierriveros committed:
change testapp location in github action

javierriveros committed:
solve conflig in .github/workflows/django.yml

javierriveros committed:
ignore db and coverage files from git

javierriveros committed:
include tox file in Dockerfile

javierriveros committed:
create named volumes for db and coverage files

javierriveros committed:
create folder for storing coverage reports

javierriveros committed:
add coverage and remove Django from requirements

javierriveros committed:
upload reports to coverage in Github Actions

EdisonArango commented on testapp/settings.py:
```diff
@@ -0,0 +1,141 @@
+    }
+}
+
+if os.environ.get('GITHUB_WORKFLOW'):
+    DATABASES = {
+        'default': {
+           'ENGINE': 'django.db.backends.postgresql',
+           'NAME': 'djangoplicity-customsearch',
+           'USER': 'djangoplicity-customsearch',
+           'PASSWORD': 'djangoplicity-customsearch',
+           'HOST': '127.0.0.1',
+           'PORT': '5432',
+        }
```
Great that you have the setup with postgres already 👍

EdisonArango commented on .github/workflows/django.yml:
```diff
@@ -0,0 +1,55 @@
+        run: python manage.py migrate
+      - name: Run tests
+        run: |
+          coverage run --source='.' manage.py test
+          coverage xml
```
If you are only using here Django tests `(manage.py test)`, then why would you need pytest?.

EdisonArango commented on tests/runtests.py:
```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env python
```
If you believe running the tests using this file with the django.setup(), then using `./manage.py test` is not required and you can instead use this file, but if you prefer the django project way using `./manage.py test` then this file is not required.

EdisonArango commented on testapp/settings.py:
```diff
@@ -0,0 +1,141 @@
+"""
+Django settings for testapp project.
```
This is not a `testapp`, this is a Django project, a Django app is a package with models, views, tests, etc.

javierriveros replied:
So, what should be the name?, This was generated by django when I created the app.

EdisonArango replied:
Again, it's not an `app`, it's a project, then better to call it: `testproject` or `test_project`

EdisonArango commented on requirements/requirements-testing.txt:
```diff
@@ -0,0 +1,5 @@
+pytest>=4.6.1,<4.7 # 4.6 is the last version to support python 2.7
+pytest-django==3.9.0
+pytest-cov==2.9.0 # coverage report
```
If you won't use pytest but only django tests then these dependencies are not required.

EdisonArango commented on docker-compose.yml:
```diff
@@ -0,0 +1,29 @@
+      - ./testapp/:/app/testapp
+      - ./htmlcov:/app/htmlcov
+      - ./.coveragerc:/app/.coveragerc
+      - ./db:/app/db
```
This db volume was intended to save a sqlite database, but as you are using postgres then it's not required, you should rather create a named volume for the postgres database.

EdisonArango commented on Makefile:
```diff
@@ -0,0 +1,2 @@
+testCoverage:
+	docker exec -it djangoplicity-customsearch-web pytest tests --cov=djangoplicity
```
Again, please keep homogeneity by running the tests the same way in all the parts, then use here the django tests instead: `./manage.py test`

javierriveros committed:
remove unused import in settings

javierriveros committed:
omit files in coverage

javierriveros committed:
remove db folder since is not required anymore

javierriveros committed:
change testenv command in tox.ini

javierriveros committed:
add pgdata volume for db service

javierriveros committed:
add additional commands to Makefile

javierriveros committed:
remove pytest from Github Actions

javierriveros committed:
remove runtests.py because we are now using the default django command

javierriveros committed:
rename testapp to test_project

testapp is a Django project, a Django app is a package with models, views, tests, etc. So, that is
why it was renamed to test_project

javierriveros committed:
remove setup.cfg

javierriveros committed:
temporarily remove python 3

javierriveros committed:
temporarily remove python 3

javierriveros committed:
add python 3 again]

javierriveros committed:
Merge remote-tracking branch 'origin/feature/tests' into feature/tests

javierriveros committed:
add python 3 again

EdisonArango committed:
add editorconfig

EdisonArango committed:
improve gitignore

EdisonArango committed:
improve docker-compose

EdisonArango committed:
using app tests

EdisonArango committed:
temp use only python 2.7

EdisonArango committed:
codecov flag
